### PR TITLE
Refacto Inference snippets tests (autogeneration + 1 snippet == 1 file)

### PR DESCRIPTION
**Pull request by Wauplin**
This PR is a refacto of the inference snippet tests. The main goal is to reduce friction when updating the inference snippets logic. In particular:
- each generated snippet is saved into a single test file with the correct file extension => should greatly improve reviews. Currently snippets are saved as hardcoded strings in JS files, making them hard to review (no syntax highlighting, weird indentation).
- snippets can be updated with a script, reducing the need to manually edit tests (which is quite time-consuming)

This PR is quite large given the auto-generated files. The main parts to review are:
- `generate-snippets-fixtures.ts` => the script that generates the snippets and test them.
- package.json, pnpm lock files, etc. => I'm not entirely sure of what I did there (especially to make `tasks-gen` depend on `tasks`)
- `python.spec.ts` / `js.specs.ts` and `curl.specs.ts` have been removed
- everything in `packages/tasks-gen/snippets-fixtures/` => the test cases. I've only committed the ones that where previously tested.

Thanks to this PR, I fixed a typo in the JS snippets (a missing comma) and curl snippets (consistency between `"` and `'`).

cc @mishig25 with whom I quickly discussed this

### CODE REVIEW

File: packages/tasks-gen/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+## @huggingface.js/tasks-gen
+
+This is package not a published one. It contains scripts to generate or test parts of the `@huggingface.js/tasks` package.
```
Comment (mishig25) [inline packages/tasks-gen/README.md:3]:
```suggestion
This package is not a published one. It contains scripts that generate or test parts of the `@huggingface.js/tasks` package.
```

File: packages/tasks-gen/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+### generate-snippets-fixtures.ts
+
+This script generates and tests Inference API snippets. The goal is to have a simple way to review changes in the snippets.
+When updating logic in `packages/tasks/src/snippets`, the test snippets must be updated and committed in the same PR.
+
+To (re-)generate the snippets, run:
+
+```
+pnpm run generate-snippets-fixtures
```
Comment (mishig25) [inline packages/tasks-gen/README.md:13]:
```suggestion
pnpm generate-snippets-fixtures
```
this is enough

File: packages/tasks-gen/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+If some logic has been updated, you should see the result with a
+```
+git diff
+# the diff has to be committed if correct
+```
+
+To test the snippets, run:
+
+```
+pnpm run test
```
Comment (mishig25) [inline packages/tasks-gen/README.md:25]:
```suggestion
pnpm test
```

File: packages/tasks-gen/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+### inference-codegen.ts
+
+Generates JS and Python dataclasses based on the Inference Specs (jsonschema files).
+
+This script is run by a cron job once a day and helps getting `@huggingface.js/tasks` and `huggingface_hub` up to date.
+
+To update the specs manually, run:
+
+```
+pnpm run inference-codegen
```
Comment (mishig25) [inline packages/tasks-gen/README.md:39]:
```suggestion
pnpm inference-codegen
```

File: packages/tasks-gen/README.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+Fetches TEI specs and generates JSON schema for input and output of text-embeddings (called feature-extraction).
+See https://huggingface.github.io/text-embeddings-inference/ for more details.
+
+This script is run by a cron job once a day and helps getting `@huggingface.js/tasks` up to date with TEI updates.
+
+To update the specs manually, run:
+
+```
+pnpm run inference-tei-import
```
Comment (mishig25) [inline packages/tasks-gen/README.md:52]:
```suggestion
pnpm inference-tei-import
```

File: packages/tasks-gen/README.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+Fetches TGI specs and generates JSON schema for input, output and stream_output of text-generation and chat-completion tasks.
+See https://huggingface.github.io/text-generation-inference/ for more details.
+
+This script is run by a cron job once a day and helps getting `@huggingface.js/tasks` up to date with TGI updates.
+
+To update the specs manually, run:
+
+```
+pnpm run inference-tgi-import
```
Comment (mishig25) [inline packages/tasks-gen/README.md:65]:
```suggestion
pnpm inference-tgi-import
```

File: packages/tasks-gen/README.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+To update the specs manually, run:
+
+```
+pnpm run inference-codegen
+```
+
+### inference-tei-import.ts
+
+Fetches TEI specs and generates JSON schema for input and output of text-embeddings (called feature-extraction).
```
Comment (mishig25) [inline packages/tasks-gen/README.md:44]:
```suggestion
Fetches TEI specs and generates JSON schema for input and output of text-embeddings (also called feature-extraction).
```

File: packages/tasks-gen/README.md (hunk 2 above)
Comment (Wauplin) [reply packages/tasks-gen/README.md:13]: TIL

File: packages/tasks-gen/scripts/generate-snippets-fixtures.ts [hunk 8]
```diff
@@ -0,0 +1,178 @@
+/*
+ * Generates Inference API snippets using @huggingface/tasks snippets.
+ *
+ * If used in test mode ("pnpm run test"), it compares the generated snippets with the expected ones.
+ * If used in generation mode ("pnpm run generate-snippets-fixtures"), it generates the expected snippets.
```
Comment (mishig25) [inline packages/tasks-gen/scripts/generate-snippets-fixtures.ts:5]:
```suggestion
 * If used in test mode ("pnpm test"), it compares the generated snippets with the expected ones.
 * If used in generation mode ("pnpm generate-snippets-fixtures"), it generates the expected snippets.
```

File: packages/tasks-gen/scripts/generate-snippets-fixtures.ts [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+ */
+
+import { existsSync as pathExists } from "node:fs";
+import * as fs from "node:fs/promises";
+import * as path from "node:path/posix";
+
+import type { InferenceSnippet } from "@huggingface/tasks";
+import { snippets } from "@huggingface/tasks";
+
+type LANGUAGE = "sh" | "js" | "py";
```
Comment (mishig25) [inline packages/tasks-gen/scripts/generate-snippets-fixtures.ts:25]: for the info, this is how the same thing is declared in moon. In subseq PR, we can probably declare `InferenceSnippetLanguage` in `tasks`and use it in `tasks-gen` & `moon` 

```ts
export const inferenceSnippetLanguages = ["python", "js", "curl"] as const;

export type InferenceSnippetLanguage = (typeof inferenceSnippetLanguages)[number];
```

File: packages/tasks-gen/scripts/generate-snippets-fixtures.ts [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+		},
+		languages: ["sh", "js", "py"],
+	},
+];
+
+const GET_SNIPPET_FN = {
+	sh: snippets.curl.getCurlInferenceSnippet,
+	js: snippets.js.getJsInferenceSnippet,
+	py: snippets.python.getPythonInferenceSnippet,
+};
```
Comment (mishig25) [inline packages/tasks-gen/scripts/generate-snippets-fixtures.ts:93]:
```suggestion
} as const;
```

File: packages/tasks-gen/scripts/generate-snippets-fixtures.ts [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+const rootDirFinder = function (): string {
```
Comment (mishig25) [inline packages/tasks-gen/scripts/generate-snippets-fixtures.ts:95]:
```suggestion
const rootDirFinder = (): string => {
```

**Review by mishig25 — APPROVED** [review]
lgtm ! 
thanks for this refacotr

**Comment by Wauplin** [comment]
thanks for the review :)

**Pull request merged**